### PR TITLE
docs: add FAQ entry about howto upgrade from Bareos < 20 with Bareos Python packages installed

### PR DESCRIPTION
**Pull request by bruno-at-bareos**
This PR add instructions to FAQ Bareos chapter about python plugin `broken packages` with `apt upgrade`

[OP #5250] [OP #5155]
#### Please check

- [X] Short description and the purpose of this PR is present _above this paragraph_
- [X] Your name is present in the AUTHORS file (optional)

If you have any questions or problems, please give a comment in the PR.

### Helpful documentation and best practices

- [Git Workflow](https://docs.bareos.org/DeveloperGuide/gitworkflow.html)
- [Automatic Sourcecode Formatting](https://docs.bareos.org/DeveloperGuide/generaldevel.html#automatic-sourcecode-formatting)
- [Check your commit messages](https://docs.bareos.org/DeveloperGuide/gitworkflow.html#commits)

##### General
- [x] PR name is meaningful
- [x] Purpose of the PR is understood
- [x] Commit descriptions are understandable and well formatted
- [x] Check backport line
- [x] Is the PR title usable as CHANGELOG entry?
- [x] Separate commit for CHANGELOG.md ("update CHANGELOG.md"). The PR number is correct.

##### Source code quality

- [x] Required documentation changes are present and part of the PR
- [x] `bareos-check-sources --since-merge` does not report any problems

### CODE REVIEW

**Review by joergsteffens — CHANGES_REQUESTED** [review]
I've tested with bareos-19.2 and packages `bareos-filedaemon` and `bareos-filedaemon-python-plugin`(and the automatic dependency `bareos-common`).

While `apt upgrade bareos-filedaemon` fails, you can work around it by:

`apt install bareos-filedaemon`

or 

`apt full-upgrade bareos-filedaemon`

will both succeed, but remove the package `bareos-filedaemon-python-plugin`.
However, if you specify 

`apt full-upgrade bareos-filedaemon bareos-filedaemon-python3-plugin`

or even only 

`apt full-upgrade bareos-filedaemon-python3-plugin`

or

`apt install bareos-filedaemon bareos-filedaemon-python3-plugin`

or

`apt install bareos-filedaemon-python3-plugin`

will all succeed and achieve what is requested.
Of course, same is true if you replace `bareos-filedaemon-python3-plugin` by `bareos-filedaemon-python2-plugin`

Given this, I'm not sure it is worth to spend a section in the Update chapter for it, as it only affects old versions. But I understood, that there have been questions about this. Would  placing the section into https://docs.bareos.org/Appendix/FAQ.html instead be helpful?

**Comment by bruno-at-bareos** [comment]
Hi @joergsteffens thanks a lot for your precious addition and tests.
I really like the 

```apt full-upgrade bareos-filedaemon bareos-filedaemon-python3-plugin``` trick and I agree that should do something with this and will more update the FAQ than the upgrade.

File: docs/manuals/source/Appendix/FAQ.rst [hunk 1]
```diff
@@ -12,6 +12,43 @@ The process for handling security-related problems is described in our GitHub `s
...
+   or been moved out of Incoming.
+   The following information may help to resolve the situation:
+
+   The following packages have unmet dependencies:
+      bareos-director-python-plugin : Depends: bareos-common (= 19.2.12-2) but 21.1.2-1 is to be installed
+      bareos-filedaemon-python-plugin : Depends: bareos-common (= 19.2.12-2) but 21.1.2-1 is to be installed
+   E: Broken packages
+
+
+In this case, it is advised to use `full-upgrade` option of apt and directly move to new recommended python3 plugin.
```
Comment (joergsteffens) [inline docs/manuals/source/Appendix/FAQ.rst:44]:
```suggestion
In this case, it is advised to use :command:`apt` with `full-upgrade` and directly move to new recommended python3 plugin.
```

File: docs/manuals/source/Appendix/FAQ.rst [hunk 2]
```diff
@@ -12,6 +12,43 @@ The process for handling security-related problems is described in our GitHub `s
 
 .. _security policy: https://github.com/bareos/bareos/security/policy
 
+.. _bareos-1920-updatefaq:
+
+Bareos 19.2 upgrade with python plugin installed
+------------------------------------------------
+
```
Comment (joergsteffens) [inline docs/manuals/source/Appendix/FAQ.rst:19]:
```suggestion

.. index::
   single: Platform; Debian
   single: bareos-19.2; Upgrade with Python packages on Debian
```

File: docs/manuals/source/Appendix/FAQ.rst [hunk 3]
```diff
@@ -12,6 +12,43 @@ The process for handling security-related problems is described in our GitHub `s
...
+On Debian platform, if you have Bareos <=19.2 with any python plugin installed, you will face some difficulties with :command:`apt upgrade` after you setup the new Bareos 20 or 21 repository.
+This is due to the renaming of generic python code to python2 packages and introducing the python3 version. dpkg is not able to handle alone this situation.
+
+
+.. code-block:: shell-session
+   :caption: Error upgrading from 19.2 with apt upgrade
+
+   apt upgrade bareos bareos-bconsole bareos-client bareos-common bareos-database-common bareos-database-postgresql bareos-database-tools bareos-director bareos-filedaemon bareos-storage bareos-storage-tape bareos-tools
```
Comment (joergsteffens) [inline docs/manuals/source/Appendix/FAQ.rst:27]:
```suggestion
   apt upgrade bareos bareos-director bareos-filedaemon
```

**Review by joergsteffens — CHANGES_REQUESTED** [review]
Looks good. I've added a few minor comments.

**Comment by bruno-at-bareos** [comment]
All changes applied, once agreed I will squash I propose to add the CHANGELOG

**Review by joergsteffens — APPROVED** [review]
Very well.

**Review by joergsteffens — APPROVED** [review]

**Pull request merged**